### PR TITLE
Fixed #26381 -- Made UserCreationForm reusable with custom user models that define USERNAME_FIELD.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/26381

### CODE REVIEW

File: django/contrib/auth/forms.py [hunk 1]
```diff
@@ -82,7 +82,9 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super(UserCreationForm, self).__init__(*args, **kwargs)
-        self.fields['username'].widget.attrs.update({'autofocus': ''})
+        username_field = self.fields.get('username')
```
Comment (charettes) [inline django/contrib/auth/forms.py]: `self.fields[self._meta.model.USERNAME_FIELD]`? I think it's a mandatory field.

File: django/contrib/auth/forms.py [hunk 2]
```diff
@@ -82,7 +82,9 @@ class Meta:
...
+        if username_field is not None:
+            self.fields['username'].widget.attrs.update({'autofocus': ''})
```
Comment (charettes) [inline django/contrib/auth/forms.py]: `username_field.widget`...

File: django/contrib/auth/forms.py (hunk 1 above)
Comment (berkerpeksag) [inline django/contrib/auth/forms.py]: I was thinking something similar to https://github.com/django/django/blob/master/django/contrib/auth/forms.py#L158, but your suggestion is better. Done, thanks!

I also removed the if branch. Perhaps we could catch the KeyError and raise an exception with a more informed message (e.g. USERNAME_FIELD cannot be found in FooCreationForm.Meta.fields)?

**Comment by charettes** [comment]
As `USERNAME_FIELD` is documented as mandatory I'm not sure it's worth raising a more descriptive exception.

**Comment by timgraham** [comment]
merged in efa9539787dbdd06cd2169023edcf7db3e2ff0c4, thanks!

**Pull request closed without merging**